diff --git a/src/thread.c b/src/thread.c
index 57f4b5aec..e7d2f40d7 100644
--- a/src/thread.c
+++ b/src/thread.c
@@ -1809,6 +1809,7 @@ void thread_detect_count(void)
 	if (!global.nbtgroups)
 		global.nbtgroups = grp_min;
 
+#if defined(USE_THREAD)
 	if (tgrp_forced && !thr_forced && !cpu_map_in_conf &&
 	    (!global.thread_limit || (global.nbthread < global.thread_limit)) &&
 	    global.nbthread < MIN(cpus_detected, thread_cpus_enabled_at_boot) &&
@@ -1828,6 +1829,7 @@ void thread_detect_count(void)
 		ha_notice("%d usable CPUs detected but 'nbthread' forced to %d (%d%%). Remove 'nbthread' for optimal tuning.\n",
 		          nbcpus, global.nbthread, global.nbthread * 100 / nbcpus);
 	}
+#endif
 
 	if (global.nbthread > global.maxthrpertgroup * global.nbtgroups) {
 		ha_diag_warning("nbthread too large or not set, found %d CPUs, limiting to %d threads (maximum is %d per thread group and %d groups). Please set nbthreads and/or increase thread-groups in the global section to silence this warning.\n",
